Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryILLPreprocess.py

```diff
@@ -397,67 +346,110 @@ def _foregroundWidths(self):
-                                     DirectBeamPosition=dbPosWS,
-                                     XUnit='TimeOfFlight',
-                                     EnableLogging=self._subalgLogging)
-                rawWS = mtd[rawWSName]
-                mergedWS = MergeRuns(InputWorkspaces=[mergedWS, rawWS],
-                                     OutputWorkspace=mergedWSName,
-                                     EnableLogging=self._subalgLogging)
-                if i == 0:
-                    self._cleanup.cleanup(firstWSName)
-                self._cleanup.cleanup(rawWS)
-            return mergedWS, beamPosWS
-        ws = self.getProperty(Prop.INPUT_WS).value
-        self._cleanup.protect(ws)
-        if not self.getProperty(Prop.BEAM_POS_WS).isDefault:
-            beamPosWS = self.getProperty(Prop.BEAM_POS_WS).value
-            self._cleanup.protect(beamPosWS)
+            loadOption = {'XUnit': 'TimeOfFlight',
+                          'BeamCentre': 0.0}
+            # MergeRunsOptions are defined by the parameter files and will not be modified here!
+            ws = LoadAndMerge(Filename=inputFiles,
+                              LoaderName='LoadILLReflectometry',
+                              LoaderVersion=-1,
+                              LoaderOptions=loadOption,
+                              OutputWorkspace=mergedWSName,
+                              EnableLogging=self._subalgLogging)
         else:
-            if not self.getProperty(Prop.BEAM_CENTRE).isDefault:
-                peakPos = self.getProperty(Prop.BEAM_CENTRE).value
-                beamPosWS = self._createFakePeakPositionTable(peakPos)
-            else:
-                # Beam position will be fitted later.
-                beamPosWS = None
-        return ws, beamPosWS
+            ws = self.getProperty(Prop.INPUT_WS).value
+            self._cleanup.protect(ws)
+        return ws
+
+    def _peakFitting(self, ws):
+        """Add peak and foreground information to the sample logs."""
+        # Convert to wavelength
+        l1 = ws.spectrumInfo().l1()
+        hists = ws.getNumberHistograms()
+        mindex = int(hists / 2)
+        l2 = ws.spectrumInfo().l2(mindex)
+        theta = ws.spectrumInfo().twoTheta(mindex) / 2.
+        linePosition = 0.0
+        if not self.getProperty(Prop.INPUT_WS).isDefault:
+            inputWS = self.getProperty(Prop.INPUT_WS).value
+            if inputWS.run().hasProperty(common.SampleLogs.LINE_POSITION):
+                linePosition = inputWS.run().getLogData(common.SampleLogs.LINE_POSITION).value
+        if not self.getProperty(Prop.LINE_POSITION_INPUT).isDefault and linePosition == 0.0:
+            linePosition = self.getProperty(Prop.LINE_POSITION_INPUT).value
+        else:
+            # Fit peak position
+            peakWSName = self._names.withSuffix('peak')
+            args = {'InputWorkspace': ws, 'OutputWorkspace': peakWSName, 'EnableLogging': self._subalgLogging}
```

Comment:
Since the `RangeLower` and `RangeUpper` need to be converted to TOF, a rewrite will result in a complication: I cannot simply user the default values (since range lower < range upper). I need to set some default ranges. Which are useful ranges? 0.0 Angstroms and 40.0 Angstroms? Better leave this to the user?

Reply:
Just pass the empty values to `FindReflectometryLines`. It handles them just fine.